### ISSUE TITLE
Mov and Heic files?

### ISSUE THREAD

**Issue by john-clark** [mjau-mjau/files.photo.gallery#22]
Mov files do not play and heic only show the type icon. Do I need to configure apache differently?

**Comment by mjau-mjau** [comment]
Unfortunately, only Safari browser supports HEIC image format, and even then, we can't create preview images from the HEIC format, because it's not supported by PHP. Similarly, I believe it's only Safari browser that supports the MOV format, although it depends slightly on the codec used.

Assuming this is media exported from an iPhone? Unfortunately, media files from iPhone are not by default web-friendly, but need to be converted. This seems a bit clumsy, but it's what Apple have decided upon.

There is nothing you can do to get other browsers to support HEIC or MOV files.

**Comment by mjau-mjau** [comment]
> https://blog.genijaho.dev/how-to-add-support-for-heic-images-with-imagemagick-in-php

Yes, it's possible if your server has both Imagemagick installed as well as PHP `exec()` available, normally only if you own your own server. Furthermore, this isn't really "support for", but it simply "converts" your HEIC images to JPG, and in this case, why not just convert them up front instead? 

If this was implemented in Files Gallery, it would need to load HEIC files through PHP `index.php?file=path/image.heic`, and then it would convert the heic to jpg. Since this would be slow, it would then need to cache converted images in `_files/images/heic/*` for faster retrieval after they are converted first time. Since the HEIC files are never loaded into browser, then why not just convert them up front instead of having them waste space and resources?

**Comment by john-clark** [comment]
HEIC are "live" images and contain more information than jpg or png or multiple frames, most people prefer to keep the source files original. You are correct that there would be load times for converting the images to thumbnails and then if clicked on a full conversion, and the need for owning the server to install prereqs. There probably is no good way to accommodate this proprietary format. I prefer the quality of the photos in this format and it would be nice if there was more support out there other than nextcloud.  Currently I have a 2nd webserver pointing at my nextcloud photos directory using your php as I prefer your layout, which works fine for photos that I have purposefully saved as an open format. Maybe just generating thumbnails with exec [vipsthumbnail ](https://stackoverflow.com/questions/12206217/efficiently-generating-thumbnails-with-imagemagick-and-convert) and then if clicked on for the full image only converting those as needed.  

All that said, I understand if this is outside the scope as you are trying for a one php file solution, which I greatly respect!

**Comment by mjau-mjau** [comment]
Hi. I appreciate your feedback regarding the HEIC format. This has already been requested earlier, with a similar response from me. HEIC is basically an Apple format, limited to support in their own Safari browser. According to caniuse.com/heif:

> **[caniuse.com/heif](https://caniuse.com/heif):** It is hard for browsers to support HEIC because it is [complex and expensive to license](https://en.wikipedia.org/wiki/High_Efficiency_Video_Coding#Patent_licensing). [AVIF](https://caniuse.com/avif) and [JPEG XL](https://caniuse.com/jpegxl) provide free licenses and are designed to supersede HEIC.

> You are correct that there would be load times for converting the images to thumbnails

Thumbnails would be the easy part, because Files Gallery already needs to resize and cache images from the other "standard" formats anyway. In the case of HEIC, we would just need to resize with Vipsthumbnail or ImageMagick via the command-line, because HEIC isn't (yet) supported by PHP.

> I prefer the quality of the photos in this format and it would be nice if there was more support out there other than nextcloud.

It's cool that Nextcloud "supports" HEIC, but just to be 100% clear, HEIC images are converted to JPG (or another browser-supported format) on output, so you will always lose the HEIC-specific quality and features of the image in browser.

This is essentially why I suggested to convert the HEIC images up front. Because ultimately, they aren't used anyway (except perhaps in Safari), and will always need to get converted. Furthermore, it will require more storage space for the original HEIC images alongside the JPG-converted copies. The only benefit of course with built-in "support", is that the conversion will be done automatically, which is very convenient of course.

> and then if clicked on for the full image only converting those as needed

I think it's worth exploring if this can be done in realtime without it being slow. After all, we are not resizing the image, just converting it to JPG for browser output. HEIC images would need to load through `index.php?file=path/image.heic` PHP proxy, which would make the request to ImageMagick, in turn serving the image. It might not be fast, but it might be worthwhile.

I will look into this later. For future requests, please post in [forum.files.gallery](https://forum.files.gallery/), as I only use Github for a repository.
